### PR TITLE
DCJ-634: Remove deprecated/unused dataset association endpoints

### PR DESCRIPTION
**Pull request by rushtong**
### Addresses
https://broadworkbench.atlassian.net/browse/DCJ-634

### Summary
Remove deprecated/unused dataset association endpoints. Historically, this was used to store the DATA OWNER role for users. This has been completely replaced with data submitters and custodians.

----
Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/main/java/org/broadinstitute/consent/http/service/UserService.java [hunk 1]
```diff
@@ -69,7 +67,7 @@ public class UserService {
   @Inject
   public UserService(UserDAO userDAO, UserPropertyDAO userPropertyDAO, UserRoleDAO userRoleDAO,
       VoteDAO voteDAO,
-      DatasetAssociationDAO datasetAssociationDAO, InstitutionDAO institutionDAO,
+      InstitutionDAO institutionDAO,
```
Comment (okotsopoulos) [inline src/main/java/org/broadinstitute/consent/http/service/UserService.java]: Not a big deal, but the formatting of this constructor declaration looks off to me, like something that would be reformatted by a linter run.

**Review by okotsopoulos — APPROVED** [review]
👍

File: src/main/java/org/broadinstitute/consent/http/service/UserService.java (hunk 1 above)
Comment (rushtong) [reply src/main/java/org/broadinstitute/consent/http/service/UserService.java]: We don't have a linter, but we do have an `.editorconfig` that applies Google's Java style rules. I'm open to adding an auto-linter to our stack - it just hasn't been a priority for us.

**Review by fboulnois — APPROVED** [review]
👍

**Pull request merged**